include/dt-bindings/pinctrl/esp32-pinctrl.h

```diff
@@ -10167,5 +10167,12 @@
 #define UART2_TX_GPIO33 \
 	ESP32_PINMUX(33, ESP_NOSIG, ESP_U2TXD_OUT)
 
+/* SMI_MDC */
+#define SMI_MDC_GPIO23 \
+	ESP32_PINMUX(23, ESP_NOSIG, ESP_EMAC_MDC_O)
+
+/* SMI_MDIO */
+#define SMI_MDIO_GPIO18 \
```

Comment:
I think we can add all the supported pins for MDC and MDIO in this list besides only GPIO23 and GPIO18. @glaubermaroto, what do you think?

Reply:
@grantramsay I understand this is only to allow the Ethernet kit to work though.

Reply:
Yes, this should for for ESP32 Ethernet Kit or Olimex ESP32-EVB / ESP32-GATEWAY boards.
Feel free to edit this PR to make it more generic

Reply:
> I think we can add all the supported pins for MDC and MDIO in this list besides only GPIO23 and GPIO18. @glaubermaroto, what do you think?

I agree, it should add all supported pins. I'll just check the documentation for possible constraints on pin selections.

Reply:
@grantramsay BTW, not sure if you are aware that these macros' definitions can also be automated: https://github.com/zephyrproject-rtos/hal_espressif/tree/zephyr/zephyr/scripts/pinctrl

Reply:
> @grantramsay BTW, not sure if you are aware that these macros' definitions can also be automated: https://github.com/zephyrproject-rtos/hal_espressif/tree/zephyr/zephyr/scripts/pinctrl
@grantramsay  It would be great if you add the pin muxing information to the pincfg file, so we don't have any issues next time the automated script run.
https://github.com/zephyrproject-rtos/hal_espressif/blob/zephyr/zephyr/esp_shared/pincfgs/esp32.yml

Reply:
The ESP32 datasheet says:

> MDC, MDIO, CRS and COL are slow signals, and can be mapped onto any GPIO pin through the GPIO-Matrix.

Maybe the `SMI_MDC_GPIOXX`/`SMI_MDIO_GPIOXX` should be moved to the board dtsi?

Reply:
Never-mind, I see that **many** defines are created when the peripheral pin locations are flexible.
I've updated the PR with these changes